.github/workflows/github.yml

```diff
@@ -138,41 +148,87 @@ jobs:
+      uses: actions/cache/save@v4
+      with:
+        path: ${{ runner.temp }}/apt-cache
+        key: ${{ steps.aptcache.outputs.cache-primary-key }}
 
     - name: Install Conan Dependencies
       if: "${{ matrix.conan_prebuilts != '' }}"
       run: source '${{github.workspace}}/CI/install_conan_dependencies.sh' '${{matrix.conan_prebuilts}}'
 
     - name: Install vcpkg Dependencies
       if: ${{ startsWith(matrix.platform, 'msvc') }}
-      run: source '${{github.workspace}}/CI/install_vcpkg_dependencies.sh' '${{matrix.platform}}'
+      run: source '${{github.workspace}}/CI/install_vcpkg_dependencies.sh' '${{matrix.platform}}' 'vcpkg'
+
+    - name: Setup MSVC Developer Command Prompt
+      if: ${{ startsWith(matrix.platform, 'msvc') }}
+      uses: ilammy/msvc-dev-cmd@v1
+      with:
+        arch: ${{ matrix.arch }}
+
+    # ensure the cache for each PR is separate so they don't interfere with each other
+    # fall back to cache of the vcmi/vcmi repo if no PR-specific cache is found
 
-    # ensure the ccache for each PR is separate so they don't interfere with each other
-    # fall back to ccache of the vcmi/vcmi repo if no PR-specific ccache is found
-    - name: Install ccache for PRs
+    - name: Setup compiler cache for PRs
       uses: hendrikmuhs/ccache-action@v1.2
-      if: ${{ github.event.number != '' && !startsWith(matrix.platform, 'msvc') }}
+      if: ${{ github.event.number != '' }}
       with:
+        variant: ${{ startsWith(matrix.platform, 'msvc') && 'sccache' || 'ccache' }}
         key: ${{ matrix.platform }}-PR-${{ github.event.number }}
         restore-keys: |
           ${{ matrix.platform }}-PR-${{ github.event.number }}
           ${{ matrix.platform }}-branch-${{ github.base_ref }}
-        max-size: "5G"
+          ${{ matrix.platform }}-
+        create-symlink: ${{ !startsWith(matrix.platform, 'msvc') }}
+        max-size: '5G'
         verbose: 2
-
-    - name: Install ccache for branch builds
+        job-summary: ""           # <-- disable built-in summary to avoid duplicate block
+        
+    - name: Setup compiler cache for branch builds
       uses: hendrikmuhs/ccache-action@v1.2
-      if: ${{ github.event.number == '' && !startsWith(matrix.platform, 'msvc')}}
+      if: ${{ github.event.number == '' }}
       with:
-        key: ${{ matrix.platform }}-${{ github.ref_name }}
+        variant: ${{ startsWith(matrix.platform, 'msvc') && 'sccache' || 'ccache' }}
+        key: ${{ matrix.platform }}-branch-${{ github.ref_name }}
         restore-keys: |
           ${{ matrix.platform }}-branch-${{ github.ref_name }}
-        max-size: "5G"
+          ${{ matrix.platform }}-
+        create-symlink: ${{ !startsWith(matrix.platform, 'msvc') }}
+        max-size: '5G'
         verbose: 2
+        job-summary: ""           # <-- disable built-in summary to avoid duplicate block
+        
+    - name: CCache tuning (Android)
+      if: ${{ startsWith(matrix.platform, 'android') }}
```

Comment:
would it make sense to enable this step for anything non-msvc?

Reply:
Probably won't change anything, but have not tested it.